rootegpythia6.sh

```
@@ -31,10 +32,21 @@ sed -i \
   -e '/^ROOT_GENERATE_DICTIONARY/i include_directories(${CMAKE_CURRENT_LIST_DIR}/inc)' \
   "$SOURCEDIR/CMakeLists.txt"
 
+# Patch the exported package config so downstream consumers (e.g. FairShip,
+# which has FairRoot's FindPythia6 on CMAKE_MODULE_PATH) get the
+# Pythia6::Pythia6 IMPORTED target that ROOTEGPythia6Targets.cmake links
+# against. FairRoot's FindPythia6 creates a non-namespaced "Pythia6" target;
+# upstream ROOTEGPythia6 expects "Pythia6::Pythia6". Define the namespaced
+# target directly so we don't depend on whichever FindPythia6 module the
+# consumer picks up first.
+sed -i 's|find_package(Pythia6 REQUIRED)|if(NOT TARGET Pythia6::Pythia6)\n  add_library(Pythia6::Pythia6 SHARED IMPORTED)\n  set_target_properties(Pythia6::Pythia6 PROPERTIES IMPORTED_LOCATION "$ENV{PYTHIA6_ROOT}/lib/libPythia6.so")\nendif()|' \
+  "$SOURCEDIR/cmake/Templates/ROOTEGPythia6Config.cmake.in"
```

Shellcheck (0.11.0)

[info] Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

(SC2317)